code/implementation/simulator/agent-connection/src/main/java/io/cattle/platform/agent/connection/simulator/impl/SimulatorStartStopProcessor.java

```diff
@@ -44,9 +45,9 @@ public Event handle(final AgentConnectionSimulator simulator, Event event) throw
         final Object uuid = CollectionUtils.getNestedValue(event.getData(), "instanceHostMap", "instance", "uuid");
         if (uuid != null) {
             if (add && !FORGET.matcher(eventString).matches()) {
-                simulator.getInstances().add(uuid.toString());
+                simulator.getInstances().put(uuid.toString(), InstanceConstants.STATE_RUNNING);
             } else {
-                simulator.getInstances().remove(uuid.toString());
+                simulator.getInstances().put(uuid.toString(), InstanceConstants.STATE_STOPPED);
             }
```

Comment:
perhaps silly question: is Stopped is the only opposite state to Running in this context? Can it be be reported as Removed for example?